components/SignInForm.tsx

```diff
@@ -0,0 +1,150 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+import { useRef, useState } from "react";
+
+// Logo Import
+import logo from "../public/Cacart_logo.png";
+import googlelogo from "../public/google_logo.png";
+
+type FormInputs = {
+  email: string;
+  password: string;
+};
+
+export default function SignInForm() {
+  // Set userData
+  const userData = useRef<FormInputs>({
+    email: "",
+    password: "",
+  });
+
+  const validateEmail = (email: string) => {
+    const re = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return re.test(email.toLowerCase());
+  };
+
+  const validatePassword = (password: string) => {
```

Comment:
@HarryLee87 Should we be validating user's password constraint from the Sign In Page? We already validate them during their sign up and I feel like this may be redundant.

Only scenario where I think it's good to add this validation is when we want to restrict user from entering password that doesn't meet our password constraint, but that doesn't seem very intuitive. 

I'd love to hear what others think about this too :)

Reply:
After thinking about what you said, I've realized that validating things like email format in SignIn may not be the best idea. In the latest commit, I've removed such validations and added empty functions that will check validity by communicating with the backend database.